src/components/app/components/Navbar.js

```diff
@@ -35,9 +43,15 @@ const items = [
  * this is the the navigation bar at the top of the home page
 */
 class Navbar extends React.Component {
+
     logout = ({ key }) => {
-        const { removeCookie } = this.props.actions;
-        key === SIGN_OUT && removeCookie();
+        const { removeCookie, resetSigninState } = this.props.actions;
+        const { isLoggedIn, isSignedUp } = this.props;
+
+        if (isLoggedIn || isSignedUp === SUCCESS) {
+            key === SIGN_OUT ? removeCookie() && 
+            message.info(VALID_SIGNOUT, 5) && resetSigninState() : "";
+        }
```

Comment:
`isLoggedIn || isSignedUp === SUCCESS` this should just be `isLoggedIn`. On successful, signup (in the `saga`) you should trigger the `action` or actions (i.e. setCookie, signinSuccess ) that "logs" on the user in on the browser.

Reply:
This should look like this:
```
    if (isLoggedIn && key === SIGN_OUT) {
        removeCookie();
        message.info(VALID_SIGNOUT, 5);
        resetSigninState();
    }
```